internal/gps/source_manager.go

```diff
@@ -148,8 +153,12 @@ var _ SourceManager = &SourceMgr{}
 // gps's SourceManager is intended to be threadsafe (if it's not, please file a
 // bug!). It should be safe to reuse across concurrent solving runs, even on
 // unrelated projects.
-func NewSourceManager(cachedir string) (*SourceMgr, error) {
-	err := os.MkdirAll(filepath.Join(cachedir, "sources"), 0777)
+func (c SourceManagerConfig) NewSourceManager() (*SourceMgr, error) {
```

Comment:
> Why not change NewSourceManager(cachedir string) into NewSourceManager(c SourceManagerConfig)?

I thought it was more verbose than it needed to be, especially for callers outside of gps:
```go
gps.NewSourceManager(gps.SourceManagerConfig{...})
```

> Or even just NewSourceManager(cachedir string, logger *log.Logger)?

There will be more parameters added in the future, and some will be optional (logger already is), so we wanted to avoid an ever expanding exported constructor.